File: C/prog1.c
#include <stdio.h>

 

int main() {

  float valores[15];
  float maior, menor;

  printf("Digite 15 valores reais:\n");
  for (int i = 0; i < 15; i++) {
    scanf("%f", &valores[i]);
  }

  maior = valores[0];
  menor = valores[0];

  for (int i = 1; i < 15; i++) {
    if (valores[i] > maior) {
      maior = valores[i];
    }

    if (valores[i] < menor) {
      menor = valores[i];
    }
  }

  printf("O maior valor é: %.2f\n", maior);
  printf("O menor valor é: %.2f\n", menor);

}

File: Java/prog1.java
import java.util.Scanner;

public class prog1 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        int val;
        do {
            System.out.print("Insira um número: ");
            val = scanner.nextInt();
            if (val != 0) {
                if (val < min) {
                    min = val;
                }
                if (val > max) {
                    max = val;
                }
            }
        } while (val != 0);
        System.out.println("O menor valor é: " + min);
        System.out.println("O maior valor é: " + max);
    }
}
